modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -409,6 +409,11 @@ Error GDScriptAnalyzer::resolve_class_inheritance(GDScriptParser::ClassNode *p_c
 				return err;
 			}
 
+			if (!parser->_is_tool && ext_parser->get_parser()->_is_tool) {
+				push_error(R"(The base class script has the "@tool" annotation, but this script does not have it.)", p_class);
+				return ERR_PARSE_ERROR;
+			}
+
```

Comment:
```suggestion
			if (!parser->_is_tool && ext_parser->get_parser()->_is_tool) {
				push_warning(R"(The base class script has the "@tool" annotation, but this script does not have it.)", p_class);
				return ERR_PARSE_ERROR;
			}

```

As discussed in the GDScript team meeting, this should return a warning instead of an error.